pkg/validation/internal/operatorhub.go

```diff
@@ -81,41 +101,202 @@ func validateBundleOperatorHub(bundle *manifests.Bundle) errors.ManifestResult {
 	for _, warn := range warns {
 		result.Add(errors.WarnInvalidCSV(warn.Error(), bundle.CSV.GetName()))
 	}
+
+	errs, warns = validateHubDeprecatedAPIS(bundle, k8sVersion)
+	for _, err := range errs {
+		result.Add(errors.ErrFailedValidation(err.Error(), bundle.CSV.GetName()))
+	}
+	for _, warn := range warns {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
+	if warn := validateHubChannels(bundle.Channels); warn != nil {
+		result.Add(errors.WarnFailedValidation(warn.Error(), bundle.CSV.GetName()))
+	}
+
 	return result
 }
 
-func validateHubCSVSpec(csv v1alpha1.ClusterServiceVersion) ([]error, []error) {
+// validateHubChannels will check the channels. The motivation for the following check is to ensure that operators
+// authors knows if their operator bundles are or not respecting the Naming Convention Rules.
+// However, the operator authors still able to choose the names as please them.
+func validateHubChannels(channels []string) error {
+	const preview = "preview"
+	const stable = "stable"
+	const fast = "fast"
+	for _, channel := range channels {
+		if !strings.Contains(channel, preview) &&
+			!strings.Contains(channel, stable) &&
+			!strings.Contains(channel, fast) {
+			return fmt.Errorf("ensure that your channels are following the channel naming convention. " +
+				"More info: https://olm.operatorframework.io/docs/best-practices/channel-naming/")
+		}
+
+	}
+	return nil
+}
+
+// validateHubDeprecatedAPIS will check if the operator bundle is using a deprecated or no longer supported k8s api
+func validateHubDeprecatedAPIS(bundle *manifests.Bundle, k8sVersion string) ([]error, []error) {
 	var errs []error
 	var warns []error
+	k8s1_22, _ := semver.Make("1.22.0")
+	k8s1_16, _ := semver.Make("1.16.0")
+	hasK8sVersion := len(k8sVersion) > 0
 
-	if csv.Spec.Provider.Name == "" {
-		errs = append(errs, fmt.Errorf("csv.Spec.Provider.Name not specified"))
+	// Transform the key/option k8sVersion in semver Version to compare
+	var semverK8sVersion semver.Version
+	if hasK8sVersion {
+		version := k8sVersion
+		if len(strings.Split(k8sVersion, ".")) == 2 {
+			// if the value found is like 1.16 then, we concat with .0 to check it with semver
+			version += ".0"
+		}
+		var err error
+		semverK8sVersion, err = semver.Make(version)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("invalid value informed via the k8s key option : %s", k8sVersion))
+		} else {
+			// we might want to return it as info instead of warning in the future.
+			warns = append(warns, fmt.Errorf("checking APIs against Kubernetes version : %s", k8sVersion))
```

Comment:
Why would you warn if version parsing was successful?

Reply:
See: https://github.com/operator-framework/api/blob/00a685b5012bc776d277eb07e77245b33ff21e16/pkg/validation/internal/operatorhub.go#L179-L184

Reply:
That isn't true. This code adds this warning regardless of whether deprecated APIs exist.

Reply:
hi @estroz, 

> This code adds this warning regardless of whether deprecated APIs exist.
Yes, we should NOT return an error in all scenarios. Please, see the comment above. It describes when the error should not be raised and why.

Reply:
I improved the explanation/comments. Also, might easier to check the tests to get better the rules.
So, I am closing this one. However, please feel free to re-open if required.

Reply:
A warning shouldn’t be returned either, since nothing is wrong with user inputs. How about just logging an info message?

Reply:
- The error/warning is returned when the operator bundle has any manifest using the deprecated APIs and the tests are considering a version >= 1.16
- However, the error is ONLY returned if the k8s flag informed or minKubeVersion is >= 1.22 and the operator bundles do NOT have any manifest with the latest api version. 

See the scenarios; 

https://github.com/operator-framework/api/blob/facca9fbef4d53e06a83e41e2183c6854bec56b3/pkg/validation/internal/operatorhub_test.go#L325-L462

Then, what is the scenario that in POV is missing here?

Reply:
@estroz thank you for your review. I think that is right now. 
also, checking further in OperatorHub will not possible either publish a project valid for both scenarios. (OLM will try to install all. ) so, it makes the tests/checks simpler as well.

Reply:
>The error/warning is returned when the operator bundle has any manifest using the deprecated APIs and the tests are considering a version >= 1.16 (we just check if has or not deprecated apis)

I'm saying that this is not true. The following summary of the current validator implementation will always return an error if `versionProvided != ""`, which effectively means that a warning will be emitted any time a user specifies a k8s version external to the bundle's `minKubeVersion`, regardless of the version specified.

```go
func validateHubDeprecatedAPIS(bundle *manifests.Bundle, versionProvided string) ([]error, []error) {
	isVersionProvided := len(versionProvided) > 0

	...

	if isVersionProvided {
		semVerVersionProvided, err = semver.ParseTolerant(versionProvided)
		if err != nil {
			errs = append(errs, fmt.Errorf("invalid value informed via the k8s key option : %s", versionProvided))
		} else {
			// we might want to return it as info instead of warning in the future.
			warns = append(warns, fmt.Errorf("checking APIs against Kubernetes version : %s", versionProvided))
		}
	}
...
}
```

Is that your intent? The tests seem to say so, and I don't think a warning should be emitted here. What if a validator consumer, like operator-sdk, were to add an option like `--fail-if-warn`? Then this validator would fail erroneously.

Reply:
Then, it means that:

--optional-values="k8s=value" flag with a value => 1.16 <= 1.22 the validator will return result as warning.
--optional-values="k8s=value" flag with a value => 1.22 the validator will return result as error.
minKubeVersion >= 1.22 return the error result.


Yes. If minKubeVersion is Empty then it means that we should raise a WARN. It is the same as 1.11+